pkg/controller/core/resourceflavor_controller.go

```diff
@@ -49,13 +59,26 @@ func (r *ResourceFlavorReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 }
 
 func (r *ResourceFlavorReconciler) Create(e event.CreateEvent) bool {
+	var toNotify bool
 	flv, match := e.Object.(*kueue.ResourceFlavor)
 	if !match {
 		return false
 	}
+	defer func() {
+		if toNotify {
+			r.notifyWatchers(flv)
+		}
+	}()
+
 	log := r.log.WithValues("resourceFlavor", klog.KObj(flv))
 	log.V(2).Info("ResourceFlavor create event")
-	r.cache.AddOrUpdateResourceFlavor(flv.DeepCopy())
+
+	// As long as one clusterQueue got ready, we should inform clusterQueue controller to broadcast the event.
+	cqNames := r.cache.AddOrUpdateResourceFlavor(flv.DeepCopy())
```

Comment:
if AddOrUpdateResourceFlavor returns a bool to indicate if at least one CQ changed its status from pending to ready, if true we call r.qManager.Broadcast(); we don't need lines 78-81 below.